src/core/steps/pipeline/pipeline.ts

```diff
@@ -40,12 +50,43 @@ export interface PipelinePerStepMetaInformation {
 	}
 }
 
+/**
+ * Returns the step with the given name from the given pipeline.
+ *
+ * @example
+ * ```ts
+ * type Foo = PipelineStepWithName<typeof DEFAULT_DATAFLOW_PIPELINE, 'parse'>
+ * // Foo contains the type of the "parse" step in the DEFAULT_DATAFLOW_PIPELINE
+ * ```
+ */
 export type PipelineStepWithName<P extends Pipeline, Name extends PipelineStepName> = P extends Pipeline<infer U> ? U extends IPipelineStep<Name> ? U : never : never
+/**
+ * Returns the processor function of the step with the given name from the given pipeline.
+ */
 export type PipelineStepProcessorWithName<P extends Pipeline, Name extends PipelineStepName> = PipelineStepWithName<P, Name>['processor']
+/**
+ * Returns the printer function of the step with the given name from the given pipeline.
+ */
 export type PipelineStepPrintersWithName<P extends Pipeline, Name extends PipelineStepName> = PipelineStepWithName<P, Name>['printer']
+/**
+ * Returns the output type of the step with the given name from the given pipeline.
+ *
+ * @example
+ * ```ts
+ * type Foo = PipelineStepOutputWithName<typeof DEFAULT_DATAFLOW_PIPELINE, 'parse'>
+ * // Foo contains the ParseStepOutput & PipelinePerStepMetaInformation type (ie the parse output and meta information)
+ * ```
+ */
 export type PipelineStepOutputWithName<P extends Pipeline, Name extends PipelineStepName> = Awaited<ReturnType<PipelineStepProcessorWithName<P, Name>>> & PipelinePerStepMetaInformation
-
-
+/**
+ * Returns a union type that represents the required inputs to be passed to the given pipeline.
+ *
+ * @example
+ * ```ts
+ * type Foo = PipelineInput<typeof DEFAULT_DATAFLOW_PIPELINE>
+ * // Foo contains ParseRequiredInput & NormalizeRequiredInput
+ * ```
```

Comment:
```suggestion
 * ```
 * 
 * In short, this can be useful whenever you want to describe _all_ inputs a complete
 * pipeline needs to run through (i.e., the union of all inputs required by the individual steps).
 *
 * @see {@link PipelineOutput}
```